omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/TaskActionResource1_8.java

```diff
@@ -0,0 +1,220 @@
+import org.openmrs.module.webservices.rest.web.annotation.PropertyGetter;
+import org.openmrs.module.webservices.rest.web.annotation.Resource;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.resource.api.Creatable;
+import org.openmrs.module.webservices.rest.web.resource.impl.BaseDelegatingResource;
+import org.openmrs.module.webservices.rest.web.resource.impl.DelegatingResourceDescription;
+import org.openmrs.module.webservices.rest.web.response.IllegalRequestException;
+import org.openmrs.module.webservices.rest.web.response.ResourceDoesNotSupportOperationException;
+import org.openmrs.module.webservices.rest.web.response.ResponseException;
+import org.openmrs.scheduler.SchedulerException;
+import org.openmrs.scheduler.TaskDefinition;
+
+import javax.servlet.ServletContext;
+import java.util.ArrayList;
+import java.util.Collection;
+
+@Resource(name = RestConstants.VERSION_1 + "/taskaction", supportedClass = TaskAction.class, supportedOpenmrsVersions = {
+        "1.8.*", "1.9.*", "1.10.*", "1.11.*", "1.12.*", "2.0.*", "2.1.*" })
+public class TaskActionResource1_8 extends BaseDelegatingResource<TaskAction> implements Creatable {
+	
+	private TaskServiceWrapper taskServiceWrapper = new TaskServiceWrapper();
+	
+	public void setTaskServiceWrapper(TaskServiceWrapper taskServiceWrapper) {
+		this.taskServiceWrapper = taskServiceWrapper;
+	}
+	
+	@Override
+	public Object create(SimpleObject post, RequestContext context) throws ResponseException {
+		TaskAction action = newDelegate();
+		setConvertedProperties(action, post, getCreatableProperties(), true);
+		
+		Collection<TaskDefinition> taskDefinitions;
+		if (action.isAllTasks() != null && action.isAllTasks()) {
+			taskDefinitions = taskServiceWrapper.getRegisteredTasks();
+			action.setTasks(new ArrayList<TaskDefinition>(taskDefinitions));
+		} else {
+			taskDefinitions = action.getTasks();
+		}
+		ServletContext servletContext = getServletContext(context);
+		
+		if (action.getAction() == TaskAction.Action.SCHEDULETASK || action.getAction() == TaskAction.Action.RESCHEDULETASK
+		        || action.getAction() == TaskAction.Action.DELETE || action.getAction() == TaskAction.Action.SHUTDOWNTASK) {
+			if (taskDefinitions == null || taskDefinitions.isEmpty()) {
+				throw new IllegalRequestException("Cannot execute action " + action.getAction()
+				        + " on empty set of task definitions.");
+			}
+		}
+		if (action.isAllTasks() == null || !action.isAllTasks()) {
+			for (TaskDefinition taskDef : taskDefinitions) {
+				if (taskDef == null) {
+					throw new IllegalRequestException(
+					        "One or more of the task definitions you specified are not registered on this server");
+				}
+			}
+		}
+		
+		switch (action.getAction()) {
+			case SCHEDULETASK:
+				scheduleTask(taskDefinitions);
+				break;
+			case RESCHEDULETASK:
+				reScheduleTask(taskDefinitions);
+				break;
```

Comment:
I understand the actions to schedule, reschedule, delete, etc. But i do not understand the actions like onstartup and onshutdown. Does the user make a rest call and say on startup? For what purpose?

Reply:
Those are default scheduler methods, 
OnStartup : Start all tasks that are scheduled to run on startup.
onShutDown : Stop all tasks and clean up the scheduler instance.

Reply:
Can you describe an existing use case where a client will make a REST call as onStartup or onShutDown?

Reply:
Putting it in a different way, let us do only the functionality that we actually need for our GSOC user interface work. No more. 😊

Reply:
If he shut down some the tasks in the system and wanted to schedule the tasks which were active in the system startup, then he can just use onStartUp function.

If he want to shut down all tasks in the system, then use onShutdow in one click

Reply:
Is that functionality needed for our GSOC user interface for managing tasks?

Reply:
Actually, It's not needed for the GSoC User interface if I remove the Shutdown all tasks button from there.
I will remove it by now and complete after the GSoC time :-)